cmd/commands/git-source.go

```diff
@@ -189,53 +199,67 @@ func NewGitSourceCreateCommand() *cobra.Command {
 				GsName:              args[1],
 				RuntimeName:         args[0],
 				CreateDemoResources: false,
+				Include:             include,
+				Exclude:             exclude,
+				Flow:                flow,
 			})
 		},
 	}
 
 	cmd.Flags().BoolVar(&createRepo, "create-repo", false, "If true, will create the specified git-source repo in case it doesn't already exist")
+	cmd.Flags().StringVar(&include, "include", "", "include")
+	cmd.Flags().StringVar(&exclude, "exclude", "", "exclude")
 
 	insCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{CloneForWrite: true})
 	gsCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{
 		Prefix:   "git-src",
 		Optional: true,
 	})
 
+	flow = store.Get().GsCreateFlow
+
 	return cmd
 }
 
 func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error {
-	// upsert git-source repo
-
-	gsRepo, gsFs, err := opts.GsCloneOpts.GetRepo(ctx)
+	version, err := getRuntimeVersion(ctx, opts.RuntimeName)
 	if err != nil {
-		return fmt.Errorf("failed to clone git-source repo: %w", err)
+		return err
 	}
 
-	if opts.CreateDemoResources {
-		if err := createDemoResources(ctx, opts, gsRepo, gsFs); err != nil {
-			return fmt.Errorf("failed to create git-source demo resources: %w", err)
-		}
+	if opts.Flow == store.Get().InstallationFlow {
+		return legacyGitSourceCreate(ctx, opts)
+	}
+
+	if version.LessThan(versionOfGitSourceByAppProxyRefactor) {
+		log.G(ctx).Infof("runtime \"%s\" is using a depracated git-source api. Versions %s and up use the app-proxy for this command", opts.RuntimeName, minAddClusterSupportedVersion)
+		return legacyGitSourceCreate(ctx, opts)
 	} else {
-		if err := createPlaceholderIfNeeded(ctx, opts, gsRepo, gsFs); err != nil {
-			return fmt.Errorf("failed to create a git-source placeholder: %w", err)
+		appProxy, err := cfConfig.NewClient().AppProxy(ctx, opts.RuntimeName, store.Get().InsecureIngressHost)
```

Comment:
Please add fallback to the `legacyGitSourceCreate` if app-proxy is not accessible